changelog/8645.improvement.rst

```diff
@@ -1,4 +1,4 @@
-Reducing confusion from `pytest.warns(None)` by:
+Reducing confusion from ``pytest.warns(None)`` by:
 
-- Allowing no arguments to be passed in order to catch any exception (no argument defaults to `Warning`).
-- Emit a deprecation warning if passed `None`.
+- Allowing no arguments to be passed in order to catch any exception (no argument defaults to :class:`Warning`).
+- Emit a deprecation warning if passed ``None``.
```

Comment:
This should probably be split off to a separate `.deprecation.rst` file? Or perhaps the whole thing should go under deprecations rather than improvements? @Zac-HD @olgarithms what do you think?

Reply:
Hmm.  I think it probably makes most sense to move this entirely to a `.deprecation.rst` file, with a message like 

```rst
:func:`pytest.warns(None) <pytest.warns>` is now deprecated because many people used
it to mean "this code does not emit warnings", but it actually had the effect of 
checking that the code emits at least one warning of any type - like ``pytest.warns()``
or ``pytest.warns(Warning)``.
```

Reply:
Happy with this decision too!